pkg/deployment/agent.go

```diff
@@ -163,8 +174,10 @@ func (a *Agent) Get() *appsv1.DaemonSet {
 							},
 							InitialDelaySeconds: 1,
 						},
-						Resources: commonSpec.Resources,
+						Resources:    commonSpec.Resources,
+						VolumeMounts: commonSpec.VolumeMounts,
 					}},
+					Volumes:            commonSpec.Volumes,
```

Comment:
The same as for sidecar, I had to add volumes and mounts to the agent container.